openpype/modules/deadline/plugins/publish/submit_publish_job.py

```diff
@@ -448,7 +448,16 @@ def _create_instances_for_aov(self, instance_data, exp_files):
             preview = False
             if app in self.aov_filter.keys():
                 for aov_pattern in self.aov_filter[app]:
-                    if re.match(aov_pattern, aov):
+                    # Matching against the AOV pattern in the render files
+                    # In order to match the AOV name
+                    # we must compare against the render filename string 
+                    # We are grabbing the render filename string
+                    # from the collection that we have grabbed from exp_files 
+                    if isinstance(col, list):
```

Comment:
The other fallback checks compare again `(list, tuple)`. Not sure if that is either too wide or this one is to narrow.
Anyway, we don't use the remainder any differently than the collection - but I do see now that the remainder branch will bug out anyway, because I believe it wouldn't have a `.head` attribute which is used [here](https://github.com/pypeclub/OpenPype/blob/80e08d43ce3617d57280a04ae426c0d42a595709/openpype/modules/deadline/plugins/publish/submit_publish_job.py#L422). Or am I mistaken?

@mkolar what's the best way to approach this? I feel the best way would be trying to work around needing all these if/else statements everywhere. Most of the code would just need a list of files looking at what it does and it's trivial to turn both cases into a valid `files` variable. E.g. `files = list(col[0])` and `files = [rem]`.

I know this PR is technically not about solving this - but the code can be greatly simplified if some time is spent fixing these issues too.

Reply:
Hey thanks man.  I opted for `list `only as I'm supposing the incoming object is a `list ` only (as that's what's being grabbed higher in the file, if I'm not mistaken).  But I'm not entirely sure to be honest whether I should be checking for `tuple `too, or not.

Also, I think there's some work being done by @antirotor on the data incoming and on a new version of this, maybe, but I have no idea as to the state nor am I 100% sure that that's even in the works.

Reply:
The check for tuple is not necessary as col will never be a tuple (it is casted to list or it should be just a string in case of single file. The reason why we don't use even single file as a list is because we still have no way in representation to differentiate between single file sequence and single file. I agree that this code needs refactoring but in connection to other issues we have down the line.

Reply:
But I think we introduced `is_sequence` tag to representation later on, so it might be obsolete. Now I am not sure. Anyway, I would leave it to different PR